### ISSUE TITLE
Resolve dynamically generated URLs assigned to AWS internal IPs/DNS automatically.

### ISSUE THREAD

**Issue by Joseph-R** [sequenceiq/cloudbreak#1025]
Hey guys,

Creating this issue here, by request from the [#cloudbreak-support-public chat](https://www.hipchat.com/gohd6v0n1).   Relates to our PoC with HDP 2.3+ and CloudBreak on AWS.

CloudBreak 1.0 creates Hadoop clusters inside a fresh VPC.  And Ambari puts Hadoop service pages (Namenode UI, Yarn Resource Manager UI, etc) on AWS external IPs.  But it looks like something is also building DNS entries off of AWS internal IPs, which are not resolvable.  

So, for example, I'm able to access the yarn resource manager at a `http://$EXTERNAL_IP:8088/cluster`, but child links to an application's history or logs will go to something like `http://ip-10-0-77-230.node.dc1.consul:8088/proxy/$APPLICATION_ID`.  Which is unreachable.  That DNS entry is created from the private IP (`10.0.77.230`) assigned to that node within the AWS VPC.  

Screenshots:
https://www.dropbox.com/s/67911dt62gi5v6l/Screenshot%202015-10-08%2013.58.23.png?dl=0
https://www.dropbox.com/s/8wzcm29vmgqxqgd/Screenshot%202015-10-08%2014.00.45.png?dl=0

The suggestions I received in the CloudBreak Public Support channel were to:
* Add the cluster ips and hostnames to your /etc/hosts file.
* SSH into one of the instances and create a a VPN connection.

I think you'd need both.  Direct access to the internal network and a means of resolving the generated DNS entries with an IP.

This is a workable patch, but it is not a great solution once you enable auto-scaling and start throwing some automation at this.  Also I am not sure how often CloudBreak will create a new VPC.   One for each cluster?  Maintaining /etc/host files at this scale is not reliable or scalable.

I'm curious what a better long-term solution would be.

**Comment by lalyos** [comment]
Hi Joe,

Maintaining an `/etc/hosts` file is not a viable option in a changing cloud environment, as you mentioned.

## consul dns
The DNS names ending with **.node.dc1.consul** are resolvable by consul, which provides service discovery and membership information on both http and DNS protocol.

You can test it easily by asking A or SRV record about the consul service itself:
```
dig @127.0.0.1 +short consul.service.consul A
dig @127.0.0.1 +short consul.service.consul SRV
```

You can resolve the mentioned node names:
```
dig @127.0.0.1 +short ip-10-0-77-230.node.dc1.consul
```

## soultion

So the solution is easy: you can set a **nameserver** pointing to consul in `/etc/resolv.conf`.
```
nameserver 127.0.0.1
```

**Comment by Joseph-R** [comment]
Hey @lalyos,

Okay, great!  I think I understand what you're saying.  That sounds like a totally viable approach when running CloudBreak locally.

Per your team's recommendation, we switched to deploying CloudBreak with one of your AMI's and running it on EC2.   I assume the same approach remains valid?  

I tried the following steps:
1. Add `nameserver  $ec2-IP` to your `/etc/resolv.conf` locally.
2. Open port 53 for both TCP/UDP (from anywhere) for DNS on the cloudbreak EC2 instance

Assuming all this to date is correct, I must still be missing a step.  I am unable to resolve `ip-10-0-77-230.node.dc1.consul` using my CloudBreak node's IP:
```
$ dig @52.23.156.69 +short ip-10-0-77-230.node.dc1.consul
;; connection timed out; no servers could be reached
```

Any feedback or "how to" on using this approach with CloudBreak on EC2?

Regards,
Joe Reid
